src/Compiler/Utilities/Caches.fs

```diff
@@ -223,13 +215,10 @@ type Cache<'Key, 'Value when 'Key: not null and 'Key: equality> internal (totalC
                     async {
                         match! mb.Receive() with
                         | EvictionQueueMessage.Add entity ->
-
-                            assert entity.Node.IsSome
-
-                            evictionQueue.AddLast(entity.Node.Value)
+                            evictionQueue.AddLast(entity.Node)
 
                             // Evict one immediately if necessary.
-                            while evictionQueue.Count > capacity do
+                            if evictionQueue.Count > capacity then
```

Comment:
Why not keeping the `while` ?

Reply:
To make it clear that we evict at most one each time we add one. With the current algorithm, an eviction failure here is in fact an internal error.